grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java

```diff
@@ -109,4 +117,33 @@ static boolean isAlive(Vertex vertex) {
         keyValues.forEachRemaining(pair -> map.put(pair.getKey(), pair.getValue()));
         return map;
     }
+
+    /**
+     * Check whether it is possible that there is a resource edge between the two given concepts.
+     */
+    public static boolean mayHaveResourceEdge(GraknTx graknGraph, ConceptId conceptId1, ConceptId conceptId2) {
+        Concept concept1 = graknGraph.getConcept(conceptId1);
+        Concept concept2 = graknGraph.getConcept(conceptId2);
+        return concept1 != null && concept2 != null && (concept1.isAttribute() || concept2.isAttribute());
+    }
+
+    /**
+     * Get the resource edge id if there is one. Return null if not.
+     */
+    public static ConceptId getResourceEdgeId(GraknTx graph, ConceptId conceptId1, ConceptId conceptId2) {
+        if (mayHaveResourceEdge(graph, conceptId1, conceptId2)) {
+            List<Answer> answers = graph.graql().match(
+                    var("x").id(conceptId1),
+                    var("y").id(conceptId2),
+                    var("z").rel(var("x")).rel(var("y"))
+            ).select("z").execute();
```

Comment:
There's a method called `get` that might help here. You can do `query.get("z").forEach(concept -> ...)`